src/browser/ui/React.js

```diff
@@ -13,6 +13,30 @@
 
 'use strict';
 
+var REACT_VERSION = '0.14.0-alpha';
+
+var ExecutionEnvironment = require('ExecutionEnvironment');
+var warning = require('warning');
+
+if (__DEV__) {
+  // before performing any initialization of React, check that
+  // only one instance is in use on the current page.
+  //
+  // Using multiple instances cause a variety of problems if elements
+  // from different versions are mixed on the same page.
+  //
+  // See issue #2402
+  if (ExecutionEnvironment.canUseDOM) {
+    warning(
+      typeof window.__REACT_VERSION__ === 'undefined',
+      'Multiple instances of React have been initialized on the same page. ' +
+      'Currently initializing React v' + REACT_VERSION + ' but another instance of React v' +
+      window.__REACT_VERSION__ + ' was already initialized'
```

Comment:
I don't think versions matter that much. The real issue is, they won't work together well, IMO it's worth explicitly stating that. Also a link to an official `fb.me` gist would be nice but that's up to maintainers.